src/javaharness/java/arcs/android/demo/service/ArcsAutofillService.java

```diff
@@ -54,7 +69,8 @@ private RemoteViews createRemoteView(String contents) {
    * from the node's autofill hint. Eventually this should talk to Arcs.
    */
   private String getAutofillSuggestion(ViewNode node) {
-    // TODO(csilvestrini): Pull autofill suggestions from Arcs.
+    // TODO(csilvestrini): Pull autofill suggestions from Arcs. Should use an AutofillParticle
+    // class, which uses the ArcsServiceBridge to communicate with the main ArcsService.
```

Comment:
this TODO isn't accurate i think, and the word "suggestions" is overloaded. 
The autofill candidates aren't pulled from the particle, instead they are treated as "rendering content" by the system. So the particle uses `output` method to send them to the Arcs service (Runtime), and Runtime uses the UiBroker and delegates the rendering content to it.
This probably means that we need an API in Arcs service to register a remote UI Broker (a callback to the Android service).
Let's discuss this in our sync next week.

Reply:
Reverted comment. Figuring out the specifics of how this should work this week SGTM